src/python/WMComponent/WorkflowUpdater/SiteListUpdater.py

```diff
@@ -0,0 +1,164 @@
+from Utils.Timers import timeFunction
+from WMCore.Agent.Harness import Harness
+from WMCore.DAOFactory import DAOFactory
+from WMCore.Services.pycurl_manager import getdata
+from WMCore.Services.WorkQueue.WorkQueue import WorkQueue
+from WMCore.WMException import WMException
+from WMCore.WMSpec.WMWorkload import WMWorkloadHelper
+from WMCore.WorkerThreads.BaseWorkerThread import BaseWorkerThread
+
+
+class SiteListUpdaterPoller(BaseWorkerThread):
+    def __init__(self, config):
+        """
+        Initialize SiteListUpdaterPoller object
+        :param config: a Configuration object with the component configuration
+        """
+        BaseWorkerThread.__init__(self)
+        myThread = threading.currentThread()
+        self.logger = myThread.logger
+
+        # the reqmgr2Url should be points to ReqMgr2 data services, i.e. /reqmgr2 end-point
+        self.wmstatsUrl = getattr(config.SiteListUpdater, "wmstatsUrl")
+
+        # provide access to WMBS in local WMAgent
+        self.daoFactory = DAOFactory(package="WMCore.WMBS",
+                                     logger=myThread.logger,
+                                     dbinterface=myThread.dbi)
+        # DB function to retrieve active workflows
+        self.listActiveWflows = self.daoFactory(classname="Workflow.GetUnfinishedWorkflows")
+
+        # local WorkQueue service
+        self.localCouchUrl = self.config.WorkQueueManager.couchurl
+        self.localWQ = WorkQueue(self.localCouchUrl,
+                                 self.config.WorkQueueManager.dbname)
+
+    def getActiveWorkflows(self):
+        """
+        Provide list of active requests within WMAgent
+        :return: dict of workflows names vs pickle files
+        """
+        # get list of active workflows in WMAgent
+        wflowSpecs = self.listActiveWflows.execute()
+
+        # construct dictionary of workflow names and their pickle files
+        wmaSpecs = {}
+        for wflowSpec in wflowSpecs:
+            name = wflowSpec['name']  # this is the name of workflow
+            pklFileName = wflowSpec['spec']  # the "spec" in WMBS table (wmbs_workflow.spec) is pkl file name
+            wmaSpecs[name] = pklFileName
+        return wmaSpecs
+
+    def getRequestSpecs(self, requests):
+        """
+        Return list of requests specs for provided list of request names
+        :param requests: list of workflow requests names
+        :return wsList: list of workflow records obtained from wmstats server, each record has the following structure
+        {"RequestName": "bla", "SiteWhitelist":[], "SiteBlacklist": []}
+        """
+        # get list of workflows from wmstats
+        states = ['running-open', 'acquired']
+        urls = []
+        for state in states:
+            url = "{}/data/filtered_requests?RequestStatus={}&mask=SiteWhitelist&mask=SiteBlacklist".format(self.wmstatsUrl, state)
+            urls.append(url)
+        response = getdata(urls, ckey(), cert())
```

Comment:
Please use the wrapper module for interacting with the WMStats Server. There is this API that reaches the endpoint you need:
https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/Services/WMStatsServer/WMStatsServer.py#L92

Reply:
done, back to sequential API processing